tests/integration/graph/test_graph_varia.py

```diff
@@ -658,3 +658,115 @@ def load_chart(n_clicks):
+            html.Button(id="button", children="Clone figure"),
+            html.Div(id="output", children=""),
+        ]
+    )
+
+    app.clientside_callback(
+        """function clone_figure(_, figure) {
+            let new_figure = {...figure};
+            let shapes = new_figure.layout.shapes || [];
+            return [new_figure, shapes.length];
+        }
+        """,
+        [Output("graph", "figure"), Output("output", "children")],
+        [Input("button", "n_clicks")],
+        [State("graph", "figure")],
+    )
+
+    dash_dcc.start_server(app)
+    button = dash_dcc.wait_for_element("#button")
+    time.sleep(1)
+
+    # Draw a shape
+    dash_dcc.click_and_hold_at_coord_fractions("#graph", 0.25, 0.25)
+    dash_dcc.move_to_coord_fractions("#graph", 0.35, 0.75)
+    dash_dcc.release()
+
+    # Click to trigger an update of the output, the shape should survive
+    dash_dcc.wait_for_text_to_equal("#output", "0")
+    button.click()
+    dash_dcc.wait_for_text_to_equal("#output", "1")
+
+    # Draw another shape
+    dash_dcc.click_and_hold_at_coord_fractions("#graph", 0.75, 0.25)
+    dash_dcc.move_to_coord_fractions("#graph", 0.85, 0.75)
+    dash_dcc.release()
+
+    # Click to trigger an update of the output, the shape should survive
+    dash_dcc.wait_for_text_to_equal("#output", "1")
+    button.click()
+    dash_dcc.wait_for_text_to_equal("#output", "2")
+
+
+def test_originals_maintained_for_responsive_override(dash_dcc):
+    # In #905 we made changes to prevent shapes from being lost.
+    # This test makes sure that the overrides applied by the `responsive`
+    # prop are "undone" when the `responsive` prop changes.
+
+    app = dash.Dash(__name__)
+
+    fig = {"data": [], "layout": {"autosize": None, "width": 300, "height": 300}}
+    graph = dcc.Graph(id="graph", figure=fig)
+
+    app.layout = html.Div(
+        [
+            graph,
+            html.Br(),
+            html.Button(id="button", children="Clone figure"),
+            html.Div(id="output", children=""),
+        ]
+    )
+
+    # Each time that the button is pressed we change responsive mode.
+    # It goes from null (default), to true, false, and back to null.
+    app.clientside_callback(
+        """function clone_figure(_, figure) {
+            let new_figure = {...figure};
+            let index = window.internal_state_905 || 0;
+            window.internal_state_905 = index + 1;
+        let responsive = [true, false, null, null][index];
+        return [new_figure, responsive, figure.layout.autosize + ' ' + figure.layout.width];
+        }
+        """,
+        [
+            Output("graph", "figure"),
+            Output("graph", "responsive"),
+            Output("output", "children"),
+        ],
+        [Input("button", "n_clicks")],
+        [State("graph", "figure")],
+    )
+
+    dash_dcc.start_server(app)
+    button = dash_dcc.wait_for_element("#button")
+    time.sleep(1)
+
+    # Draw a shape
+    dash_dcc.release()
```

Comment:
why does just `release` create a shape? And where is this releasing? Should this have `click_and_hold_at_coord_fractions` and `move_to_coord_fractions` like above?

Reply:
I think this is a leftover that I forgot to delete after copying from the other example :D